odiglet/pkg/kube/instrumentation_ebpf/instrumentationconfig.go

```diff
@@ -37,40 +48,162 @@ func (i *InstrumentationConfigReconciler) Reconcile(ctx context.Context, req ctr
+		return ctrl.Result{}, instrumentationRequestErr
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (i *InstrumentationConfigReconciler) sendConfigUpdates(ctx context.Context, podWorkload k8sconsts.PodWorkload, instrumentationConfig *odigosv1.InstrumentationConfig) error {
+	if i.ConfigUpdates == nil {
+		return nil
+	}
+
+	if len(instrumentationConfig.Spec.SdkConfigs) == 0 {
+		return nil
+	}
+
+	// send a config update request for all the instrumentation which are part of the workload.
+	// if the config request is sent, the configuration updates will occur asynchronously.
+	ctx, cancel := context.WithTimeout(ctx, configUpdateTimeout)
+	defer cancel()
+
+	configUpdate := instrumentation.ConfigUpdate[ebpf.K8sConfigGroup]{}
+	for _, sdkConfig := range instrumentationConfig.Spec.SdkConfigs {
+		cg := ebpf.K8sConfigGroup{Pw: podWorkload, Lang: sdkConfig.Language}
+		currentConfig := sdkConfig
+		configUpdate[cg] = &currentConfig
+	}
+
+	select {
+	case i.ConfigUpdates <- configUpdate:
+		return nil
+	case <-ctx.Done():
+		if ctx.Err() == context.DeadlineExceeded {
+			// returning the error to retry the reconciliation
+			return errConfigUpdateTimeout
 		}
+		return ctx.Err()
+	}
+}
 
-		// send a config update request for all the instrumentation which are part of the workload.
-		// if the config request is sent, the configuration updates will occur asynchronously.
-		ctx, cancel := context.WithTimeout(ctx, configUpdateTimeout)
-		defer cancel()
+// sendInstrumentationRequest sends an instrumentation request for all processes that are part of the given workload
+// and run in containers that support instrumentation without a restart.
+func (i *InstrumentationConfigReconciler) sendInstrumentationRequest(ctx context.Context, podWorkload k8sconsts.PodWorkload, instrumentationConfig *odigosv1.InstrumentationConfig) error {
+	logger := log.FromContext(ctx)
 
-		configUpdate := instrumentation.ConfigUpdate[ebpf.K8sConfigGroup]{}
-		for _, sdkConfig := range instrumentationConfig.Spec.SdkConfigs {
-			cg := ebpf.K8sConfigGroup{Pw: podWorkload, Lang: sdkConfig.Language}
-			currentConfig := sdkConfig
-			configUpdate[cg] = &currentConfig
+	// check for distributions that support instrumentation without a restart
+	distroByContainer := make(map[string]*distro.OtelDistro)
+	for _, containerConfig := range instrumentationConfig.Spec.Containers {
+		d := i.DistributionGetter.GetDistroByName(containerConfig.OtelDistroName)
+		if d != nil && d.RuntimeAgent != nil && d.RuntimeAgent.NoRestartRequired {
+			distroByContainer[containerConfig.ContainerName] = d
 		}
+	}
 
-		select {
-		case i.ConfigUpdates <- configUpdate:
-			return ctrl.Result{}, nil
-		case <-ctx.Done():
-			if ctx.Err() == context.DeadlineExceeded {
-				// returning the error to retry the reconciliation
-				return ctrl.Result{}, errConfigUpdateTimeout
+	// if none of the containers support instrumentation without a restart - nothing to do here
+	if len(distroByContainer) == 0 {
+		return nil
+	}
```

Comment:
if a container was agent enabled true, and changed to false, it will not get uninstrumented here